hbase_storage/src/main/java/com/splicemachine/access/client/SkeletonClientSideRegionScanner.java

```diff
@@ -68,6 +68,8 @@
     private boolean flushed;
     private long numberOfRows = 0;
     private FileSystem customFilesystem;
+    private List<Cell> rowBuffer = null;
+    private boolean noMoreRecords = false;
```

Comment:
Is this field necessary? could you please consider removing it, it makes understanding the code unnecessarily more complicated IMHO, i.e.:
make `fetchNextAndGiveBuffer()` return boolean indicating whether more rows are to come  (similar to `RegionScanner.nextRaw`) and assign it to a local variable in `nextRaw` call (maybe with some documentation)

Reply:
That's not possible. The attribute would be read only the next time that nextRaw is being called, and that happens through some hbase code.